oauth2_http/java/com/google/auth/oauth2/IdTokenCredentials.java

```diff
@@ -0,0 +1,276 @@
+ * Cloud Functions or Cloud Run.<br>
+ * The following Credential subclasses support IDTokens: ServiceAccountCredentials,
+ * ComputeEngineCredentials, ImpersonatedCredentials.
+ *
+ * <p>For more information see <br>
+ * Usage:<br>
+ *
+ * <pre>
+ * String credPath = "/path/to/svc_account.json";
+ * String targetAudience = "https://example.com";
+ *
+ * // For Application Default Credentials (as ServiceAccountCredentials)
+ * // export GOOGLE_APPLICATION_CREDENTIALS=/path/to/svc.json
+ * GoogleCredentials adcCreds = GoogleCredentials.getApplicationDefault();
+ * IdTokenCredentials tokenCredential = IdTokenCredentials.create(adcCreds, targetAudience);
+ *
+ * // for ServiceAccountCredentials
+ * ServiceAccountCredentials saCreds = ServiceAccountCredentials.fromStream(new FileInputStream(credPath));
+ * saCreds = (ServiceAccountCredentials) saCreds.createScoped(Arrays.asList("https://www.googleapis.com/auth/iam"));
+ * IdTokenCredentials tokenCredential = IdTokenCredentials.create(saCreds, targetAudience);
+ *
+ * // for ComputeEngineCredentials
+ * ComputeEngineCredentials caCreds = ComputeEngineCredentials.create();
+ * IdTokenCredentials tokenCredential = IdTokenCredentials.create(caCreds, targetAudience,
+ *     Arrays.asList(ComputeEngineCredentials.ID_TOKEN_FORMAT_FULL));
+ *
+ * // for ImpersonatedCredentials
+ * ImpersonatedCredentials imCreds = ImpersonatedCredentials.create(saCreds,
+ *     "impersonated-account@project.iam.gserviceaccount.com", null,
+ *     Arrays.asList("https://www.googleapis.com/auth/cloud-platform"), 300);
+ * IdTokenCredentials tokenCredential = IdTokenCredentials.create(imCreds, targetAudience,
+ *     Arrays.asList(ImpersonatedCredentials.INCLUDE_EMAIL));
+ *
+ * // Use the IdTokenCredential in an authorized transport
+ * GenericUrl genericUrl = new GenericUrl("https://example.com");
+ * HttpCredentialsAdapter adapter = new HttpCredentialsAdapter(tokenCredential);
+ * HttpTransport transport = new NetHttpTransport();
+ * HttpRequest request = transport.createRequestFactory(adapter).buildGetRequest(genericUrl);
+ * HttpResponse response = request.execute();
+ *
+ * // Print the token, expiration and the audience
+ * System.out.println(tokenCredential.getIdToken().getTokenValue());
+ * System.out.println(tokenCredential.getIdToken().getJsonWebSignature().getPayload().getAudienceAsList());
+ * System.out.println(tokenCredential.getIdToken().getJsonWebSignature().getPayload().getExpirationTimeSeconds());
+ * </pre>
+ */
+public class IdTokenCredentials extends OAuth2Credentials {
+
+  private static final long serialVersionUID = -2133257318957588431L;
+  private static final String CLOUD_PLATFORM_SCOPE =
+      "https://www.googleapis.com/auth/cloud-platform";
+
+  private GoogleCredentials sourceCredentials;
+  private final String transportFactoryClassName;
+  private String targetAudience;
+  private List<IdTokenProvider.Option> options;
+
+  private transient HttpTransportFactory transportFactory;
```

Comment:
This looks unused - we are delegating the fetching to the source `GoogleCredentials` (which should be `IdTokenProvider` instead). We can remove this and the `transportFactoryClassName`.

Reply:
removed `transportFactoryClassName`  but see previous comment on using `IdTokenProvider` instead of `GoogleCredentials`